nix-shell fails to set up `devEnv_39` on macOS

Filing here to capture the issue.

When entering `nix-shell` from `main` (f362192) with clang 16, `devEnv_39` fails to build successfully.

The issue appears to be related to `ruamel-yaml-clib`. See:
- https://sourceforge.net/p/ruamel-yaml-clib/tickets/22/
- https://sourceforge.net/p/ruamel-yaml-clib/tickets/32/

Some patches exist on those tickets.

`nix log` output for the dependency:

<details>

```shellsession
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing pip-build-hook
Using pipBuildPhase
Using pipShellHook
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/8p1xkp4gvy60c4v5if546wgppny4r8g0-ruamel.yaml.clib-0.2.8.tar.gz
source root is ruamel.yaml.clib-0.2.8
setting SOURCE_DATE_EPOCH to timestamp 1696319303 of file ruamel.yaml.clib-0.2.8/setup.cfg
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
Removing path dependencies
Finished removing path dependencies
Removing git dependencies
Finished removing git dependencies
Removing wheel-url dependencies
Finished removing wheel-url dependencies
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
Executing pipBuildPhase
Creating a wheel...
[33mWARNING: The directory '/homeless-shelter/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.[0m[33m
[0mProcessing /private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/ruamel.yaml.clib-0.2.8
  Running command Preparing metadata (pyproject.toml)
  running dist_info
  creating /private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/pip-modern-metadata-oxi93rz7/ruamel.yaml.clib.egg-info
  writing /private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/pip-modern-metadata-oxi93rz7/ruamel.yaml.clib.egg-info/PKG-INFO
  writing dependency_links to /private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/pip-modern-metadata-oxi93rz7/ruamel.yaml.clib.egg-info/dependency_links.txt
  writing top-level names to /private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/pip-modern-metadata-oxi93rz7/ruamel.yaml.clib.egg-info/top_level.txt
  writing manifest file '/private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/pip-modern-metadata-oxi93rz7/ruamel.yaml.clib.egg-info/SOURCES.txt'
  reading manifest file '/private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/pip-modern-metadata-oxi93rz7/ruamel.yaml.clib.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  adding license file 'LICENSE'
  writing manifest file '/private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/pip-modern-metadata-oxi93rz7/ruamel.yaml.clib.egg-info/SOURCES.txt'
  creating '/private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/pip-modern-metadata-oxi93rz7/ruamel.yaml.clib-0.2.8.dist-info'
  Preparing metadata (pyproject.toml) ... [?25l[?25hdone
Building wheels for collected packages: ruamel.yaml.clib
  Running command Building wheel for ruamel.yaml.clib (pyproject.toml)
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-11.0-arm64-cpython-39
  creating build/lib.macosx-11.0-arm64-cpython-39/ruamel
  creating build/lib.macosx-11.0-arm64-cpython-39/ruamel/yaml
  creating build/lib.macosx-11.0-arm64-cpython-39/ruamel/yaml/clib
  copying ./__init__.py -> build/lib.macosx-11.0-arm64-cpython-39/ruamel/yaml/clib
  copying ./setup.py -> build/lib.macosx-11.0-arm64-cpython-39/ruamel/yaml/clib
  copying ./LICENSE -> build/lib.macosx-11.0-arm64-cpython-39/ruamel/yaml/clib
  running build_ext
  building '_ruamel_yaml' extension
  creating build/temp.macosx-11.0-arm64-cpython-39
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/nix/store/fmzf0y9bwg9jfns065f971lfvv7bkbs3-libxcrypt-4.4.36/include -I/nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9 -c _ruamel_yaml.c -o build/temp.macosx-11.0-arm64-cpython-39/_ruamel_yaml.o
  In file included from _ruamel_yaml.c:1149:
  ./_ruamel_yaml.h:10:9: warning: 'PyString_CheckExact' macro redefined [-Wmacro-redefined]
  #define PyString_CheckExact PyBytes_CheckExact
          ^
  _ruamel_yaml.c:998:11: note: previous definition is here
    #define PyString_CheckExact          PyUnicode_CheckExact
            ^
  _ruamel_yaml.c:4156:17: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
    __pyx_v_value = yaml_get_version_string();
                  ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:4462:3: warning: code will never be executed [-Wunreachable-code]
    goto __pyx_L3_error;
    ^~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:4619:3: warning: code will never be executed [-Wunreachable-code]
    goto __pyx_L3_error;
    ^~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:5069:3: warning: code will never be executed [-Wunreachable-code]
    goto __pyx_L3_error;
    ^~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:5410:3: warning: code will never be executed [-Wunreachable-code]
    goto __pyx_L3_error;
    ^~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:5536:3: warning: code will never be executed [-Wunreachable-code]
    goto __pyx_L3_error;
    ^~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:5929:52: error: incompatible function pointer types passing 'int (void *, char *, int, int *)' to parameter of type 'yaml_read_handler_t *' (aka 'int (*)(void *, unsigned char *, unsigned long, unsigned long *)') [-Wincompatible-function-pointer-types]
      yaml_parser_set_input((&__pyx_v_self->parser), __pyx_f_6ruamel_4yaml_4clib_12_ruamel_yaml_input_handler, ((void *)__pyx_v_self));
                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ./yaml.h:1370:30: note: passing argument to parameter 'handler' here
          yaml_read_handler_t *handler, void *data);
                               ^
  _ruamel_yaml.c:6170:59: warning: passing 'char *' to parameter of type 'const unsigned char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      yaml_parser_set_input_string((&__pyx_v_self->parser), PyString_AS_STRING(__pyx_v_stream), PyString_GET_SIZE(__pyx_v_stream));
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ./_ruamel_yaml.h:11:29: note: expanded from macro 'PyString_AS_STRING'
  #define PyString_AS_STRING  PyBytes_AS_STRING
                              ^
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/cpython/bytesobject.h:32:31: note: expanded from macro 'PyBytes_AS_STRING'
  #define PyBytes_AS_STRING(op) (assert(PyBytes_Check(op)), \
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ./yaml.h:1344:30: note: passing argument to parameter 'input' here
          const unsigned char *input, size_t size);
                               ^
  _ruamel_yaml.c:7887:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.tag_directive.handle); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 416, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:7899:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.tag_directive.prefix); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 417, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:8572:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.alias.value); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 447, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:8630:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.anchor.value); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 450, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:8688:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.tag.handle); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 453, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:8700:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.tag.suffix); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 454, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:8797:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_DecodeUTF8(__pyx_v_token->data.scalar.value, __pyx_v_token->data.scalar.length, ((char *)"strict")); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 459, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:452:17: note: passing argument to parameter 'string' here
      const char *string,         /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:10545:42: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
          __pyx_t_4 = PyUnicode_FromString(__pyx_v_tag_directive->handle); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 573, __pyx_L1_error)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:10557:42: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
          __pyx_t_4 = PyUnicode_FromString(__pyx_v_tag_directive->prefix); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 574, __pyx_L1_error)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:10740:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.alias.anchor); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 585, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:10818:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.scalar.anchor); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 590, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:10859:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.scalar.tag); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 593, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:10880:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_4 = PyUnicode_DecodeUTF8(__pyx_v_event->data.scalar.value, __pyx_v_event->data.scalar.length, ((char *)"strict")); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 594, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:452:17: note: passing argument to parameter 'string' here
      const char *string,         /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:11182:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.sequence_start.anchor); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 619, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:11223:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.sequence_start.tag); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 622, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:11419:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.mapping_start.anchor); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 636, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:11460:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.mapping_start.tag); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 639, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:13221:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_3 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.alias.anchor); if (unlikely(!__pyx_t_3)) __PYX_ERR(0, 733, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:13482:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_5 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.scalar.anchor); if (unlikely(!__pyx_t_5)) __PYX_ERR(0, 749, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:13538:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_5 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.sequence_start.anchor); if (unlikely(!__pyx_t_5)) __PYX_ERR(0, 752, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:13594:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_5 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.mapping_start.anchor); if (unlikely(!__pyx_t_5)) __PYX_ERR(0, 755, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:14176:36: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
    __pyx_t_2 = PyUnicode_DecodeUTF8(__pyx_v_self->parsed_event.data.scalar.value, __pyx_v_self->parsed_event.data.scalar.length, ((char *)"strict")); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 790, __pyx_L1_error)
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:452:17: note: passing argument to parameter 'string' here
      const char *string,         /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:14371:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.scalar.tag); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 803, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:14850:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_3 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.sequence_start.tag); if (unlikely(!__pyx_t_3)) __PYX_ERR(0, 836, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:15415:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_3 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.mapping_start.tag); if (unlikely(!__pyx_t_3)) __PYX_ERR(0, 875, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/include/python3.9/unicodeobject.h:138:17: note: passing argument to parameter 'u' here
      const char *u              /* UTF-8 encoded string */
                  ^
  _ruamel_yaml.c:16882:53: error: incompatible function pointer types passing 'int (void *, char *, int)' to parameter of type 'yaml_write_handler_t *' (aka 'int (*)(void *, unsigned char *, unsigned long)') [-Wincompatible-function-pointer-types]
    yaml_emitter_set_output((&__pyx_v_self->emitter), __pyx_f_6ruamel_4yaml_4clib_12_ruamel_yaml_output_handler, ((void *)__pyx_v_self));
                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ./yaml.h:1837:31: note: passing argument to parameter 'handler' here
          yaml_write_handler_t *handler, void *data);
                                ^
  _ruamel_yaml.c:18282:44: warning: assigning to 'yaml_char_t *' (aka 'unsigned char *') from 'char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
          __pyx_v_tag_directives_end->handle = PyString_AS_STRING(__pyx_v_handle);
                                             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:18397:44: warning: assigning to 'yaml_char_t *' (aka 'unsigned char *') from 'char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
          __pyx_v_tag_directives_end->prefix = PyString_AS_STRING(__pyx_v_prefix);
                                             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:18732:61: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_alias_event_initialize(__pyx_v_event, __pyx_v_anchor) == 0);
                                                              ^~~~~~~~~~~~~~
  ./yaml.h:555:63: note: passing argument to parameter 'anchor' here
  yaml_alias_event_initialize(yaml_event_t *event, yaml_char_t *anchor);
                                                                ^
  _ruamel_yaml.c:19443:62: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_scalar_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0);
                                                               ^~~~~~~~~~~~~~
  ./yaml.h:581:22: note: passing argument to parameter 'anchor' here
          yaml_char_t *anchor, yaml_char_t *tag,
                       ^
  _ruamel_yaml.c:19443:78: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_scalar_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0);
                                                                               ^~~~~~~~~~~
  ./yaml.h:581:43: note: passing argument to parameter 'tag' here
          yaml_char_t *anchor, yaml_char_t *tag,
                                            ^
  _ruamel_yaml.c:19443:91: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_scalar_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0);
                                                                                            ^~~~~~~~~~~~~
  ./yaml.h:582:22: note: passing argument to parameter 'value' here
          yaml_char_t *value, int length,
                       ^
  _ruamel_yaml.c:19874:70: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_sequence_start_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_sequence_style) == 0);
                                                                       ^~~~~~~~~~~~~~
  ./yaml.h:604:22: note: passing argument to parameter 'anchor' here
          yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                       ^
  _ruamel_yaml.c:19874:86: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_sequence_start_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_sequence_style) == 0);
                                                                                       ^~~~~~~~~~~
  ./yaml.h:604:43: note: passing argument to parameter 'tag' here
          yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                                            ^
  _ruamel_yaml.c:20305:69: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_mapping_start_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_mapping_style) == 0);
                                                                      ^~~~~~~~~~~~~~
  ./yaml.h:636:22: note: passing argument to parameter 'anchor' here
          yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                       ^
  _ruamel_yaml.c:20305:85: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_mapping_start_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_mapping_style) == 0);
                                                                                      ^~~~~~~~~~~
  ./yaml.h:636:43: note: passing argument to parameter 'tag' here
          yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                                            ^
  _ruamel_yaml.c:22198:42: warning: assigning to 'yaml_char_t *' (aka 'unsigned char *') from 'char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_v_tag_directives_end->handle = PyString_AS_STRING(__pyx_v_handle);
                                           ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:22313:42: warning: assigning to 'yaml_char_t *' (aka 'unsigned char *') from 'char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_v_tag_directives_end->prefix = PyString_AS_STRING(__pyx_v_prefix);
                                           ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  _ruamel_yaml.c:23168:64: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
      __pyx_t_2 = (yaml_alias_event_initialize((&__pyx_v_event), __pyx_v_anchor) == 0);
                                                                 ^~~~~~~~~~~~~~
  ./yaml.h:555:63: note: passing argument to parameter 'anchor' here
  yaml_alias_event_initialize(yaml_event_t *event, yaml_char_t *anchor);
                                                                ^
  _ruamel_yaml.c:23898:67: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_2 = (yaml_scalar_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0);
                                                                    ^~~~~~~~~~~~~~
  ./yaml.h:581:22: note: passing argument to parameter 'anchor' here
          yaml_char_t *anchor, yaml_char_t *tag,
                       ^
  _ruamel_yaml.c:23898:83: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_2 = (yaml_scalar_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0);
                                                                                    ^~~~~~~~~~~
  ./yaml.h:581:43: note: passing argument to parameter 'tag' here
          yaml_char_t *anchor, yaml_char_t *tag,
                                            ^
  _ruamel_yaml.c:23898:96: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_2 = (yaml_scalar_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0);
                                                                                                 ^~~~~~~~~~~~~
  ./yaml.h:582:22: note: passing argument to parameter 'value' here
          yaml_char_t *value, int length,
                       ^
  _ruamel_yaml.c:24255:75: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_2 = (yaml_sequence_start_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_sequence_style) == 0);
                                                                            ^~~~~~~~~~~~~~
  ./yaml.h:604:22: note: passing argument to parameter 'anchor' here
          yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                       ^
  _ruamel_yaml.c:24255:91: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_2 = (yaml_sequence_start_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_sequence_style) == 0);
                                                                                            ^~~~~~~~~~~
  ./yaml.h:604:43: note: passing argument to parameter 'tag' here
          yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                                            ^
  _ruamel_yaml.c:24755:74: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_2 = (yaml_mapping_start_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_mapping_style) == 0);
                                                                           ^~~~~~~~~~~~~~
  ./yaml.h:636:22: note: passing argument to parameter 'anchor' here
          yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                       ^
  _ruamel_yaml.c:24755:90: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        __pyx_t_2 = (yaml_mapping_start_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_mapping_style) == 0);
                                                                                           ^~~~~~~~~~~
  ./yaml.h:636:43: note: passing argument to parameter 'tag' here
          yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                                            ^
  53 warnings and 2 errors generated.
  error: command '/nix/store/6kpjydf9x7zqa1xq2qipnwr32ki3fs2n-clang-wrapper-16.0.6/bin/clang' failed with exit code 1
  [1;31merror[0m: [1msubprocess-exited-with-error[0m
  
  [31m×[0m [32mBuilding wheel for ruamel.yaml.clib [0m[1;32m([0m[32mpyproject.toml[0m[1;32m)[0m did not run successfully.
  [31m│[0m exit code: [1;36m1[0m
  [31m╰─>[0m See above for output.
  
  [1;35mnote[0m: This error originates from a subprocess, and is likely not a problem with pip.
  [1;35mfull command[0m: [34m/nix/store/p74zff3wk6mcrs0y8vr848yhj6nxic8h-python3-3.9.18/bin/python3.9 /nix/store/1wmz7304b4lwpjkiscy1kmlzvla9d5wj-python3.9-pip-23.3.1/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_wheel /private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/tmpak2jgjjt[0m
  [1;35mcwd[0m: /private/tmp/nix-build-python3.9-ruamel-yaml-clib-0.2.8.drv-0/ruamel.yaml.clib-0.2.8
  Building wheel for ruamel.yaml.clib (pyproject.toml) ... [?25l[?25herror
[31m  ERROR: Failed building wheel for ruamel.yaml.clib[0m[31m
[0mFailed to build ruamel.yaml.clib
[31mERROR: Failed to build one or more wheels[0m[31m
[0m
```

</details>


An easy local workaround is to comment out `devEnv_39` from `pkgs.mkShell.buildInputs`, but that's not a great long-term option.

I tried to figure out a way to force the 3.9 build to use an older version of `clang`, but my nix understanding still isn't very clear yet.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -63,8 +63,8 @@ jobs:
           name: htmlcov
           path: htmlcov/
 
-  test_39:
-    name: "Python 3.9 Tests"
+  test_310:
+    name: "Python 3.10 Tests"
 
     runs-on: ubuntu-22.04
 
@@ -74,29 +74,31 @@ jobs:
         with:
           cachix_auth_token: '${{ secrets.CACHIX_AUTH_TOKEN }}'
 
-      - name: unit tests
+      - name: Run linters and unit tests
         run: |
-          nix-shell --run "PYTHON=python3.9 make unit"
+          nix-shell --run "PYTHON=python3.10 make lint all=true"
+          nix-shell --run "PYTHON=python3.10 make types"
+          nix-shell --run "PYTHON=python3.10 make unit"
 
       - name: Run tests for the singlefile example
         run: |
           cd examples/singlefile
-          nix-shell --run "python3.9 -m unittest app.py"
+          nix-shell --run "python3.10 -m unittest app.py"
 
       - name: Run tests for the todoapp example
         run: |
           cd examples/todoapp
-          nix-shell --run "python3.9 -m unittest tests.py"
+          nix-shell --run "python3.10 -m unittest tests.py"
 
       - name: Run tests for the splitfile example
         run: |
           cd examples/splitfile
-          nix-shell --run "python3.9 -m unittest tests.py"
+          nix-shell --run "python3.10 -m unittest tests.py"
 
       - name: Save coverage report
         uses: actions/upload-artifact@v4
         with:
-          name: htmlcov-py39
+          name: htmlcov-py310
           path: htmlcov/
 
 
@@ -104,7 +106,7 @@ jobs:
     name: "Release to PyPI"
 
     if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
-    needs: [test_312, test_39]
+    needs: [test_312, test_310]
     runs-on: ubuntu-22.04
 
     # To test publishing to testpypi:

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -2,6 +2,14 @@ default_stages: [commit,push]
 repos:
 -   repo: local
     hooks:
+    -   id: pyupgrade
+        name: pyupgrade
+        description: Upgrades syntax for modern versions of Python.
+        entry: pyupgrade
+        language: system
+        types: [python]
+        args: [--py310-plus]
+
     -   id: isort
         name: isort
         description: A Python utility that sorts imports alphabetically.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -28,5 +28,5 @@ In CI, we want to test the oldest supported versions of `openapi-core` and `pyra
 * Have the `py39` folder with additional `pyproject.toml` and `poetry.lock` files
   that are changed to pin `openapi-core` and `pyramid` to minimally supported version.
 * They are auto-generated when running `make lock`.
-* They are used by Nix to prepare the Python 3.9 env.
-* `PYTHON=python3.9 make tests` then run tests with an older Python version.
+* They are used by Nix to prepare the Python 3.10 env.
+* `PYTHON=python3.10 make tests` then run tests with an older Python version.

--- a/Makefile
+++ b/Makefile
@@ -17,9 +17,9 @@ lock:
 	@cat pyproject.toml \
 		| sed 's/openapi-core = ">=/openapi-core = "==/g' \
 		| sed 's/pyramid = ">=/pyramid = "==/g' \
-		> py39/pyproject.toml
+		> py310/pyproject.toml
 	@rm -rf .venv/
-	@poetry lock --no-update --directory py39
+	@poetry lock --no-update --directory py310
 	@rm -rf .venv/
 	@nix-shell --run true
 	@direnv reload

--- a/README.md
+++ b/README.md
@@ -180,7 +180,7 @@ The authors of pyramid_openapi3 believe that the approach of validating a manual
 
 ## Running tests
 
-You need to have [poetry](https://python-poetry.org/) and Python 3.9 & 3.12 installed on your machine. All `Makefile` commands assume you have the Poetry environment activated, i.e. `poetry shell`.
+You need to have [poetry](https://python-poetry.org/) and Python 3.10 & 3.12 installed on your machine. All `Makefile` commands assume you have the Poetry environment activated, i.e. `poetry shell`.
 
 Alternatively, if you use [nix](https://nix.dev/tutorials/declarative-and-reproducible-developer-environments), run `nix-shell` to drop into a shell that has everything prepared for development.
 
@@ -206,7 +206,7 @@ These packages tackle the same problem-space:
 
 We do our best to follow the rules below.
 
-- Support the latest few releases of Python, currently Python 3.9 through 3.12.
+- Support the latest few releases of Python, currently Python 3.10 through 3.12.
 - Support the latest few releases of Pyramid, currently 1.10.7 through 2.0.2.
 - Support the latest few releases of `openapi-core`, currently just 0.19.0.
 - See `poetry.lock` for a frozen-in-time known-good-set of all dependencies.

--- a/examples/singlefile/README.md
+++ b/examples/singlefile/README.md
@@ -7,7 +7,7 @@ The `app.py` file in this folder showcases how to use the [pyramid_openapi3](htt
 ```bash
 * git clone https://github.com/Pylons/pyramid_openapi3.git
 * cd pyramid_openapi3/examples/singlefile
-* virtualenv -p python3.9 .
+* virtualenv -p python3.10 .
 * source bin/activate
 * pip install pyramid_openapi3
 * python app.py

--- a/examples/splitfile/README.md
+++ b/examples/splitfile/README.md
@@ -7,7 +7,7 @@ This folder showcases how to use the [pyramid_openapi3](https://github.com/Pylon
 ```bash
 $ git clone https://github.com/Pylons/pyramid_openapi3.git
 $ cd pyramid_openapi3/examples/splitfile
-$ virtualenv -p python3.9 .
+$ virtualenv -p python3.10 .
 $ source bin/activate
 $ pip install pyramid_openapi3
 $ python app.py

--- a/examples/todoapp/README.md
+++ b/examples/todoapp/README.md
@@ -7,7 +7,7 @@ This folder showcases how to use the [pyramid_openapi3](https://github.com/Pylon
 ```bash
 $ git clone https://github.com/Pylons/pyramid_openapi3.git
 $ cd pyramid_openapi3/examples/todoapp
-$ virtualenv -p python3.9 .
+$ virtualenv -p python3.10 .
 $ source bin/activate
 $ pip install pyramid_openapi3
 $ python app.py

--- a/py310/pyproject.toml
+++ b/py310/pyproject.toml
@@ -24,7 +24,7 @@ exclude = ["pyramid_openapi3/tests/"]
 
 
 [tool.poetry.dependencies]
-python = "^3.9"
+python = "^3.10"
 
 openapi-core = "==0.19.1"
 pyramid = "==1.10.7"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -24,7 +24,7 @@ exclude = ["pyramid_openapi3/tests/"]
 
 
 [tool.poetry.dependencies]
-python = "^3.9"
+python = "^3.10"
 
 openapi-core = ">=0.19.1"
 pyramid = ">=1.10.7"

--- a/pyramid_openapi3/exceptions.py
+++ b/pyramid_openapi3/exceptions.py
@@ -153,4 +153,4 @@ class MissingEndpointsError(Exception):
     def __init__(self, missing: t.List[str]) -> None:
         self.missing = missing
         message = f"Unable to find routes for endpoints: {', '.join(missing)}"
-        super(MissingEndpointsError, self).__init__(message)
+        super().__init__(message)

--- a/pyramid_openapi3/tests/test_permissions.py
+++ b/pyramid_openapi3/tests/test_permissions.py
@@ -19,7 +19,7 @@ DEFAULT_ACL = [
 ]
 
 
-class DummyDefaultContext(object):  # noqa: D101
+class DummyDefaultContext:  # noqa: D101
 
     __acl__ = DEFAULT_ACL
 

--- a/pyramid_openapi3/tests/test_validation.py
+++ b/pyramid_openapi3/tests/test_validation.py
@@ -28,7 +28,7 @@ class DummyRoute:  # noqa: D101
     pattern: str
 
 
-class DummyStartResponse(object):  # noqa: D101
+class DummyStartResponse:  # noqa: D101
     def __call__(self, status: str, headerlist: t.List[t.Tuple[str, str]]) -> None:
         """WSGI start_response protocol."""
         self.status = status

--- a/pyramid_openapi3/tests/test_views.py
+++ b/pyramid_openapi3/tests/test_views.py
@@ -22,7 +22,7 @@ import tempfile
 import typing as t
 
 
-class DummyStartResponse(object):  # noqa: D101
+class DummyStartResponse:  # noqa: D101
     def __call__(self, status: str, headerlist: t.List[t.Tuple[str, str]]) -> None:
         """WSGI start_response protocol."""
         self.status = status

--- a/shell.nix
+++ b/shell.nix
@@ -29,10 +29,10 @@ let
     python = pkgs.python312;
   });
 
-  devEnv_39 = poetry2nix.mkPoetryEnv (commonPoetryArgs // {
-    python = pkgs.python39;
-    pyproject = ./py39/pyproject.toml;
-    poetrylock = ./py39/poetry.lock;
+  devEnv_310 = poetry2nix.mkPoetryEnv (commonPoetryArgs // {
+    python = pkgs.python310;
+    pyproject = ./py310/pyproject.toml;
+    poetrylock = ./py310/poetry.lock;
   });
 
 in
@@ -42,7 +42,7 @@ pkgs.mkShell {
 
   buildInputs = with pkgs; [
     devEnv_312
-    devEnv_39
+    devEnv_310
     poetry
     gitAndTools.pre-commit
     nixpkgs-fmt

